ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java

```diff
@@ -317,12 +319,32 @@ public String convert(String type) {
     if (type == null) {
       return null;
     }
+    
+    type = extract(type);
+    
     if (type.contains("[]")) {
       return convertArrayType(type);
     }
     return typeConverter.convert(type);
   }
 
+  // if columnType is different for different platforms, use pattern
+  // @Column(columnDefinition = PLATFORM1;DEFINITION1;PLATFORM2;DEFINITON2;DEFINITON-DEFAULT)
+  // e.g. @Column(columnDefinition = "db2;blob(64M);sqlserver,h2;varchar(227);varchar(127)")
+  private String extract(String type) {
+    String[] tmp = type.split(";");
+    assert tmp.length % 2 == 1;
```

Comment:
```suggestion
    assert tmp.length % 2 == 1 : "You need an odd number of arguments. See Issue #2559 for details";
```
maybe throw an IAE (asserts may be disabled)